### DISCUSSION TITLE
local preset

### DISCUSSION THREAD

**Discussion by Dingchunyuan** [milaboratory/mixcr#1825] [General]
## Checklist before submitting the issue:

 - [√ ] The issue is strongly related to the MiXCR software
 - [ √] The issue can be reproduced with [the most recent version](https://github.com/milaboratory/mixcr/releases) of MiXCR
 - [√ ] There is no answer to the question in the [official documentation](http://mixcr.readthedocs.io/) and there is no duplicate issue in the [bug tracker](https://github.com/milaboratory/mixcr/issues?utf8=✓&q=)  
 - [ ] Inspection of raw alignments with [exportAlignmentsPretty](http://mixcr.readthedocs.io/en/master/export.html#exporting-well-formatted-alignments-for-manual-inspection) shows that data has the expected architecture, and sample preparation artefacts are not the reason of the problem (if this is the matter of the issue)

Hello,

I would like to ask where I should place the file if I need to set up local presets. Are there any naming requirements for this file?

I have tried placing it in the path where mixcr and micr.jar are located, the path set by the fq file, and using bashrc to set environment variables, but all attempts have failed. Below is my code.
![image](https://github.com/user-attachments/assets/306c33d6-f1fa-4715-b60c-db16fdb2ff04)
![image](https://github.com/user-attachments/assets/cbbb6651-a875-40cb-93be-80fdb60288d7)

Thank you for your assistance.

**Comment by mizraelson** [comment]
Hi, you can put it either inside the directory you run mixcr from or in ~/.mixcr/presets folder.

Then run as:

```
mixcr analyze local:preset_dcy ...
```

**Reply by Dingchunyuan** [reply]
Hello,

Thank you for your reply. However, I encountered the following error when running it. I used the following code to create the preset and only modified the values for `relativeMinScore`, `maxHits`, and `absoluteMinScore`.

**Code to create the preset:**
```bash
mixcr exportPreset --preset-name generic-amplicon-with-umi \
    --species hs \
    -f \
    --rna \
    --library imgt \
    --rigid-left-alignment-boundary \
    --floating-right-alignment-boundary C \
    --tag-pattern "^(R1:*)\^(UMI:N{16})(R2:*)" \
    --assemble-clonotypes-by CDR3 \
    preset_dcy.yaml 
 ```

**the code I ran**
``` bash 
mixcr analyze local:preset_dcy \
    S009_{{R}}_L{{n}}.fq.gz \
	./results/S009 \
    -f \
    --keep-non-CDR3-alignments \
    --rigid-left-alignment-boundary \
    --floating-right-alignment-boundary C \
    --tag-pattern "^(R1:*)\^(UMI:N{16})(R2:*)" 
```

**error**
```
picocli.CommandLine$ExecutionException: Error while running command analyze com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "gs" (class com.milaboratory.mixcr.presets.Presets$MiXCRParamsBundleRaw), not marked as ignorable (27 known properties: "exportAlignments", "inheritFrom", "flags", "vendor", "resolvedVendor$delegate", "exportClones", "refineTagsAndSort", "mixins", "resolvedCategory$delegate", "extend", "assembleCells", "abstract", "assembleContigs", "deprecation", "rawParent$delegate", "validation", "exportCloneGroups", "category", "resolvedMixins$delegate", "align", "assemblePartial", "assemble", "resolvedFlags$delegate", "pipeline", "label", "mitool", "qc"])
 at [Source: (File); line: 323, column: 1] (through reference chain: com.milaboratory.mixcr.presets.Presets$MiXCRParamsBundleRaw["gs"])
```

I noticed that this issue might be related to the problem I mentioned in [issue #1819](https://github.com/milaboratory/mixcr/issues/1819), where the code runs without these parameters but fails when they are added.
